main.go

```diff
@@ -353,135 +356,124 @@ func _main(ctx context.Context) error {
+		} // otherwise likely err == sql.ErrNoRows
+	}
 
-		// Allow WiredDB/stakedb to catch up to the auxDB, but after
-		// fetchToHeight, WiredDB must receive block signals from auxDB, and
-		// stakedb must send connect signals to auxDB.
-		fetchToHeight = lastBlockPG + 1
-
-		// For consistency with StakeDatabase, a non-negative height is needed.
-		heightDB := lastBlockPG
-		if heightDB < 0 {
-			heightDB = 0
-		}
+	// Get the last block added to the aux DB.
+	lastBlockPG, err := pgDB.HeightDB()
+	if err != nil && err != sql.ErrNoRows {
+		return fmt.Errorf("Unable to get height from PostgreSQL DB: %v", err)
+	}
 
-		// Aux DB height and stakedb height must be equal. StakeDatabase will
-		// catch up automatically if it is behind, but we must rewind it here if
-		// it is ahead of auxDB. For auxDB to receive notification from
-		// StakeDatabase when the required blocks are connected, the
-		// StakeDatabase must be at the same height or lower than auxDB.
-		stakedbHeight := int64(baseDB.GetStakeDB().Height())
-		if stakedbHeight > heightDB {
-			// Have baseDB rewind it's the StakeDatabase. stakedbHeight is
-			// always rewound to a height of zero even when lastBlockPG is -1,
-			// hence we rewind to heightDB.
-			log.Infof("Rewinding StakeDatabase from block %d to %d.",
-				stakedbHeight, heightDB)
-			stakedbHeight, err = baseDB.RewindStakeDB(ctx, heightDB)
-			if err != nil {
-				return fmt.Errorf("RewindStakeDB failed: %v", err)
-			}
+	// Allow WiredDB/stakedb to catch up to the pgDB, but after
+	// fetchToHeight, WiredDB must receive block signals from pgDB, and
+	// stakedb must send connect signals to pgDB.
+	fetchToHeight := lastBlockPG + 1
 
-			// Verify that the StakeDatabase is at the intended height.
-			if stakedbHeight != heightDB {
-				return fmt.Errorf("failed to rewind stakedb: got %d, expecting %d",
-					stakedbHeight, heightDB)
-			}
-		}
+	// For consistency with StakeDatabase, a non-negative height is needed.
+	heightDB := lastBlockPG
+	if heightDB < 0 {
+		heightDB = 0
+	}
 
-		// TODO: just use getblockchaininfo to see if it still syncing and what
-		// height the network's best block is at.
-		blockHash, nodeHeight, err := dcrdClient.GetBestBlock()
+	// Aux DB height and stakedb height must be equal. StakeDatabase will
+	// catch up automatically if it is behind, but we must rewind it here if
+	// it is ahead of pgDB. For pgDB to receive notification from
+	// StakeDatabase when the required blocks are connected, the
+	// StakeDatabase must be at the same height or lower than pgDB.
+	stakedbHeight := int64(stakeDB.Height())
+	if stakedbHeight > heightDB {
+		// Have baseDB rewind it's the StakeDatabase. stakedbHeight is
+		// always rewound to a height of zero even when lastBlockPG is -1,
+		// hence we rewind to heightDB.
+		log.Infof("Rewinding StakeDatabase from block %d to %d.",
+			stakedbHeight, heightDB)
+		stakedbHeight, err = baseDB.RewindStakeDB(ctx, heightDB)
```

Comment:
Note that stakeDB has a `Rewind` function of it's own, but `WiredDB`'s version has logging and context cancellation.